File: scrimage-core/src/test/scala/com/sksamuel/scrimage/Issue54Test.scala
package com.sksamuel.scrimage

import org.scalatest.{FunSuite, Matchers}

/** @author Stephen Samuel */
class Issue54Test extends FunSuite with Matchers {

  test("image for issue 54 should load as expected") {
    ImmutableImage.fromStream(getClass.getResourceAsStream("/issue54.jpg")).width shouldBe 2560
    ImmutableImage.fromStream(getClass.getResourceAsStream("/issue54.jpg")).scaleToWidth(400) shouldBe
      ImmutableImage.fromStream(getClass.getResourceAsStream("/issue54.png"))
  }
}

File: scrimage-core/src/test/kotlin/com/sksamuel/scrimage/core/Issue54Test.kt
package com.sksamuel.scrimage.core

import com.sksamuel.scrimage.ImmutableImage
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class Issue54Test : FunSpec({

   test("image for issue 54 should load as expected") {
      ImmutableImage.loader().fromResource("/issue54.jpg").width shouldBe 2560
      ImmutableImage.loader().fromResource("/issue54.jpg").scaleToWidth(400).width shouldBe 400
   }

})
